content/docs/ops/security/keys-and-certs/index.md

```diff
@@ -179,3 +181,87 @@ Certificate:
          0c:33:42:f0:b6:f5:1c:0d:fb:70:65:aa:93:3e:ca:0e:58:ec:
          8e:d5:d0:1e
 {{< /text >}}
+
+## How Citadel decides whether to create Service Account secrets
+
+### Overview
+
+When a Citadel instance notices that a `ServiceAccount` was created in a namespace, it needs to decide whether it should generate an `istio.io/key-and-cert` secret for that `ServiceAccount`. In order to make that decision, there are three relevant values:
+
+1. **`ca.istio.io/env` namespace label**: *string valued* label which should be pointed at the namespace of the desired Citadel instance
+
+1. **`ca.istio.io/override` namespace label**: *boolean valued* label which overrides all other configurations and forces all Citadel instances either to target or ignore a namespace
+
+1. [**`enableNamespacesByDefault` security configuration**](/docs/reference/config/installation-options/#security-options): default behavior if no labels are found on the `ServiceAccount's` namespace
+
+From these three values, the decision process mirrors that of the [`Sidecar Injection Webhook`](/docs/ops/setup/injection/). The detailed behavior is that:
+
+* If `ca.istio.io/override` exists and is valid, then follow what it tells us
+* If not, then check `ca.istio.io/env`. If the value matches the Citadel instance's namespace, the namespace should be targeted
+* If `ca.istio.io/env` is nonexistent or invalid, follow the `enableNamespacesByDefault` configuration value
+* If `enableNamespacesByDefault` is `true`, the Citadel instance should create secrets for unlabeled namespaces, otherwise it should not
+
+This logic is captured in the truth table below:
+
+| `ca.istio.io/override` value | `ca.istio.io/env` match | `enableNamespacesByDefault` configuration | Workload secret created |
+|------------------------------|-------------------------|-------------------------------------------|-------------------------|
+|true|yes|true|yes|
+|true|yes|false|yes|
+|true|no|true|yes|
+|true|no|false|yes|
+|true|`unset`|true|yes|
+|true|`unset`|false|yes|
+|false|yes|true|no|
+|false|yes|false|no|
+|false|no|true|no|
+|false|no|false|no|
+|false|`unset`|true|no|
+|false|`unset`|false|no|
+|`unset`|yes|true|yes|
+|`unset`|yes|false|yes|
+|`unset`|no|true|no|
+|`unset`|no|false|no|
+|`unset`|`unset`|true|yes|
+|`unset`|`unset`|false|no|
+
+{{< idea >}}
+When a namespace transitions from `disabled` to `enabled`, Citadel will retroactively generate secrets for all `ServiceAccounts` in that namespace. When transitioning from `enabled` to `disabled`, however, Citadel will not delete the namespace's generated secrets until certificate renewal.
+{{< /idea >}}
+
+### Quick examples
```

Comment:
We only have `kubectl label ns`, I'm thinking complete instructions to walk through people E2E will be better.

1. create ns foo, bar
1. install citadel in foo, install citadel with another helm options in bar.
1. kubectl to verify the results in different ns are taken care by different citadel.

I mean otherwise, customers understand the concept and mechanism, but still have to combine the details and config knob themselves.

@geeknoid @rcaballeromx thoughts?

Reply:
I see your point, E2E examples would be better. At that point we might have to move this from the `Operations` doc section to the `Tasks` doc section (when it becomes more of a tutorial-type doc, creating namespaces from ground up, and walking users through etc).